contracts/SportMarkets/LiquidityPool/AMMLiquidityPool.sol

```diff
@@ -0,0 +1,511 @@
+        initNonReentrant();
+        sportsAMM = ISportsAMM(params._sportsAmm);
+
+        sUSD = params._sUSD;
+        roundLength = params._roundLength;
+        maxAllowedDeposit = params._maxAllowedDeposit;
+        minDepositAmount = params._minDepositAmount;
+        maxAllowedUsers = params._maxAllowedUsers;
+
+        sUSD.approve(address(sportsAMM), type(uint256).max);
+    }
+
+    /// @notice Start pool and begin round #1
+    function start() external onlyOwner {
+        require(!started, "Liquidity pool has already started");
+        require(allocationPerRound[1] > 0, "can not start with 0 deposits");
+        round = 1;
+        firstRoundStartTime = block.timestamp;
+        started = true;
+    }
+
+    /// @notice Deposit funds from user into pool for the next round
+    /// @param amount Value to be deposited
+    function deposit(uint amount) external canDeposit(amount) nonReentrant whenNotPaused {
+        uint nextRound = round + 1;
+        address roundPool = _getOrCreateRoundPool(nextRound);
+        sUSD.safeTransferFrom(msg.sender, roundPool, amount);
+
+        if (!whitelistedDeposits[msg.sender]) {
+            require(
+                (balancesPerRound[round][msg.sender] + amount) <=
+                    ((stakingThales.stakedBalanceOf(msg.sender) * stakedThalesMultiplier) / ONE),
+                "Not enough staked THALES"
+            );
+        }
+
+        require(msg.sender != defaultLiquidityProvider, "Can't deposit directly as default liquidity provider");
+
+        // new user enters the pool
+        if (balancesPerRound[round][msg.sender] == 0 && balancesPerRound[nextRound][msg.sender] == 0) {
+            require(usersCurrentlyInPool < maxAllowedUsers, "Max amount of users reached");
+            usersPerRound[nextRound].push(msg.sender);
+            userInRound[nextRound][msg.sender] = true;
+            usersCurrentlyInPool = usersCurrentlyInPool + 1;
+        }
+
+        balancesPerRound[nextRound][msg.sender] += amount;
+
+        allocationPerRound[nextRound] += amount;
+        totalDeposited += amount;
+
+        if (address(stakingThales) != address(0)) {
+            stakingThales.updateVolume(msg.sender, amount);
+        }
+
+        emit Deposited(msg.sender, amount, round);
+    }
+
+    function _depositAsDefault(
+        uint amount,
+        address roundPool,
+        uint _round
+    ) internal {
+        require(defaultLiquidityProvider != address(0), "default liquidity provider not set");
+
+        sUSD.safeTransferFrom(defaultLiquidityProvider, roundPool, amount);
+
+        balancesPerRound[_round][defaultLiquidityProvider] += amount;
+        allocationPerRound[_round] += amount;
+
+        emit Deposited(defaultLiquidityProvider, amount, _round);
+    }
+
+    function commitTrade(
+        address market,
+        uint sUSDAmount,
+        ISportsAMM.Position position
+    ) external nonReentrant whenNotPaused onlyAMM returns (address liquidityPoolRound) {
```

Comment:
The usual input parameters are:
- market
- position
- amount/buyAmount

Not to make confusion in integration with UI, other contracts.